File: Problem Set 2/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main()
{
    string str = get_string("Text: ");
    int l = 0, w = 1, s = 0;
    for (int i = 0; i < strlen(str); i++)
    {
        if (isalnum(str[i]))
        {
            l++;
        }
        if (isspace(str[i]))
        {
            w++;
        }
        if (str[i] == '.' || str[i] == '!' || str[i] == '?')
        {
            s++;
        }
    }
    double L = 100 * (double) l / w;
    double S = 100 * (double) s / w;
    double index = 0.0588 * L - 0.296 * S - 15.8;
    if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int)round(index));
    }
}

File: Python/Problem Set 6/Readability/readability.py
import cs50

st = cs50.get_string("Text: ")
l = 0
w = 1
s = 0

for i in range(len(st)):
    if (st[i].isalnum()):
        l += 1
    if (st[i].isspace()):
        w += 1
    if (st[i] == '.' or st[i] == '!' or st[i] == '?'):
        s += 1
        
L = 100 * l / w
S = 100 * s / w
index = 0.0588 * L - 0.296 * S - 15.8

if (index > 16):
    print("Grade 16+")
elif (index < 1):
    print("Before Grade 1")
else:
    print("Grade "+str((round(index))))
